src/main/cpp/src/genomicsdb/variant_storage_manager.cc

```diff
@@ -293,6 +293,68 @@ void VariantArrayInfo::write_cell(const void* ptr) {
   m_buffer_offsets[coords_buffer_idx] += coords_size;
 }
 
+int read_row_bounds_kernel(TileDB_CTX *ctx,
+                                              rapidjson::Document *json_doc, 
+                                              char **buffer,
+                                              const std::string& filepath) {
+  size_t size = file_size(ctx, filepath);
+  if (size > 0) {
+    *buffer = (char *)malloc(size+1);
+    if (!*buffer)
+      throw VariantStorageManagerException(std::string("Out-of-memory exception while allocating memory"));
+    memset(*buffer, 0, size+1);
+
+    if (read_file(ctx, filepath, 0, *buffer, size) ||
+      close_file(ctx, filepath)) {
+      free(*buffer);
+      return -1;
+    }
+
+    json_doc->Parse(*buffer);
+    if (json_doc->HasParseError()) {
+      free(*buffer);
+      return -1;
+    }
+    return 0;
+  }
+  return -1;
+}
+
+// special case setup of tiledb ctx for JNI accesses that may need it directly
```

Comment:
This seems to be replicated code. Is it possible to use intialize_workspace() tiledb_utils.cc from the tiledb api?
```
/**
 * Returns 0 when workspace is created
 *        -1 when path is not a directory
 *        -2 when workspace could not be created
 *         1 when workspace exists and nothing is changed
 */
#define OK 0
#define NOT_DIR -1
#define NOT_CREATED -2
#define UNCHANGED 1
int initialize_workspace(TileDB_CTX **ptiledb_ctx, const std::string& workspace, const bool replace, const bool disable_file_locking)
```

Reply:
Ohh - I didn't know about that. Will switch to using that...